googletest/include/gtest/internal/gtest-port.h

```diff
@@ -677,6 +679,17 @@ typedef struct _RTL_CRITICAL_SECTION GTEST_CRITICAL_SECTION;
   type(type const &) = delete; \
   GTEST_DISALLOW_ASSIGN_(type)
 
+// A macro to disallow move operator=
+// This should be used in the private: declarations for a class.
```

Comment:
@trzecieu Hi Piotr, for me as a GTest user, I think your pull request could certainly be useful to get rid of those clang-tidy warnings. I would just like to mention that the current [Google Style Guide](https://google.github.io/styleguide/cppguide.html#Copyable_Movable_Types) says that those deleted member functions should be in the `public` section, instead of being declared `private`! Furthermore I noted that the code examples suggested that `operator=` should return a reference to the type, rather than `void`  I leave it up to you if you want to address these things, these are just my 2 cents. Kind regards, Niels

Reply:
Thx, it's a good suggestion that I'll follow. Also I've found 2 possible smells in this area. Will come back with further improvements

Reply:
Should be fixed now. With some asterisks, that I'm unable to test code as I'm in travel at the moment. Will continue verification once I settle somewhere for a while :) Added also 2 more modernizations, explained in the main description. 
Thanks!